packages/synapse-react-client/src/utils/functions/FileHandleUtils.ts

```diff
@@ -72,12 +80,22 @@ export function getFileHandleStorageInfo(
  * @param fileHandle
  * @returns
  */
-export function getStorageLocationName(fileHandle: FileHandle) {
+export function getStorageLocationName(
+  fileHandle: FileHandle,
+  storageLocationUploadDestination?: UploadDestination,
+) {
+  const baseKey =
+    storageLocationUploadDestination &&
+    (storageLocationUploadDestination as BaseKeyUploadDestination).baseKey
+  const baseKeyText = baseKey ? `/${baseKey}` : ''
+
   switch (fileHandle.concreteType) {
     case PROXY_FILE_HANDLE_CONCRETE_TYPE_VALUE:
       return (fileHandle as ProxyFileHandle).filePath
     case EXTERNAL_OBJECT_STORE_FILE_HANDLE_CONCRETE_TYPE_VALUE:
-      return (fileHandle as ExternalObjectStoreFileHandle).bucket
+      return `${
+        (fileHandle as ExternalObjectStoreFileHandle).bucket
+      }${baseKeyText}`
```

Comment:
@nickgros Do you know whether the value of bucket in `fileHandle` always match the value in `storageLocationUploadDestination`? Would it be appropriate to call [`getUploadDestinationString`](https://github.com/Sage-Bionetworks/synapse-web-monorepo/blob/a4d978ba1de54da55ece084c75d077fefbb3562c/packages/synapse-react-client/src/utils/functions/FileHandleUtils.ts#L123) on the `storageLocationUploadDestination` instead?

Reply:
I am not 100% sure but I would expect it does always match the value (the backend probably JOINs this data when we request it). These implementations are looking very similar, so I think it would be a good idea to reuse them.

Reply:
Are there cases where a storage location may not have an upload destination?

Reply:
> Are there cases where a storage location may not have some related upload destination?

If a `storageLocationId` exists, then the [service](https://rest-docs.synapse.org/rest/GET/entity/id/uploadDestination/storageLocationId.html) will always return an upload destination. However, `storageLocationId` is optional in the [FileHandle](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/FileHandle.html). Based on [this comment](https://github.com/Sage-Bionetworks/synapse-web-monorepo/blob/a4d978ba1de54da55ece084c75d077fefbb3562c/packages/synapse-react-client/src/utils/functions/FileHandleUtils.ts#L84), when `storageLocationId` is `null`, we assume the file handle is  in Synapse Storage. So even if use `getUploadDestinationString` to deduplicate some logic, we would still need to handle this case. Are there other cases you're considering that we'd need to handle?

Reply:
I was thinking of [ExternalFileHandles](https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/file/ExternalFileHandle.html) but those are unaffected by this change